algebird-core/src/main/scala/com/twitter/algebird/QTree.scala

```diff
@@ -63,8 +61,26 @@ object QTree {
   def apply(k: Double): QTree[Double] = apply(k -> k)
 }
 
-class QTreeSemigroup[A: Monoid](k: Int) extends Semigroup[QTree[A]] {
+class QTreeSemigroup[A](k: Int)(implicit val underlyingMonoid: Monoid[A]) extends Semigroup[QTree[A]] {
```

Comment:
So you can have access to it if you need it. For instance, a number of methods in QTree now require a `Monoid[A]`, If you get passed only a QTreeSemigroup, you could not call them, so you can by getting the monoid out.

That's the reasoning. Maybe we can do better.

One idea is to make those methods actually require the QTreeSemigroup, which does not run the risk that somehow a user will set up incoherent instances of the Monoids (more than one in play depending on call site). What do you think?